arkouda/io.py

```diff
@@ -444,24 +446,43 @@ def _parse_obj(obj: Dict) -> Union[Strings, pdarray, arkouda.array_view.ArrayVie
         flat = create_pdarray(components[0])
         shape = create_pdarray(components[1])
         return arkouda.array_view.ArrayView(flat, shape)
+    elif "categorical" == obj["arkouda_type"]:
+        return Categorical.from_return_msg(obj["created"])
     else:
         raise TypeError(f"Unknown arkouda type:{obj['arkouda_type']}")
 
 
-def _dict_recombine_segarrays(df_dict):
+def _dict_recombine_segarrays_categoricals(df_dict):
     # this assumes segments will always have corresponding values.
     # This should happen due to save config
     seg_cols = ["_".join(col.split("_")[:-1]) for col in df_dict.keys() if col.endswith("_segments")]
-    df_dict_keys = [
-        "_".join(col.split("_")[:-1]) if col.endswith("_segments") or col.endswith("_values") else col
+    cat_cols = ["_".join(col.split(".")[:-1]) for col in df_dict.keys() if col.endswith(".categories")]
```

Comment:
I could be wrong but I think this should be 
```python
cat_cols = [".".join(col.split(".")[:-1]) for col in df_dict.keys() if col.endswith(".categories")]
```

Reply:
It should not matter, but I am seeing that if there is another `.` in the name it could cause and issue. I will update that. Thanks!

Reply:
Updated